### ISSUE TITLE
t.test with factors should use factor labels instead of estimate1/2

### ISSUE THREAD

**Issue by jankatins** [tidymodels/broom#115]
Example:

``` R
df <- data.frame(a=1:8, b=factor(c(0,1,0,1,0,1,0,1), levels=c(0,1), labels=c("a", "b")) )
print(df %>% t.test(a~b, data = .))
print(df %>% t.test(a~b, data = .) %>% tidy())
```

results in 

```
    Welch Two Sample t-test

data:  a by b
t = -0.54772, df = 6, p-value = 0.6036
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -5.467429  3.467429
sample estimates:
mean in group a mean in group b      ######### <- factor labels
              4               5 

  estimate estimate1 estimate2  statistic   p.value parameter  conf.low     ####### <- no factor labels :-(
1       -1         4         5 -0.5477226 0.6036451         6 -5.467429
  conf.high
1  3.467429
```

**Comment by alexpghayes** [comment]
I think this makes using a single t-test object nicer. It would make working with lists of t-tests much harder, however -- the following would break due to differing column names:

```r
purrr::map_dfr(list_of_tests, tidy)
```

**Comment by alexpghayes** [comment]
I like this! I think having `x` and `y` as group names isn't ideal, but it's the best workaround and we can document it carefully.

**Comment by nutterb** [comment]
Included in #326
